pkg/api/networkservice/mechanisms/wireguard/helpers.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wireguard
+
+import (
+	"net"
+	"strconv"
+
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+)
+
+// Mechanism - a wireguard mechanism utility wrapper
+type Mechanism interface {
+	// SrcIP -  src ip
+	SrcIP() net.IP
+	// DstIP - dst ip
+	DstIP() net.IP
+	// SrcPublicKey - source public key
+	SrcPublicKey() (string, error)
+	// DstPublicKey - destination public key
+	DstPublicKey() (string, error)
+	// SrcPrivateKey - source private key
+	SrcPrivateKey() (string, error)
+	// dstPrivateKey - destination private key
+	DstPrivateKey() (string, error)
```

Comment:
Should we be passing Private Keys in our messages?

Reply:
Looks like with a new architecture we can generate these keys on cross-connect NSE side and not put it into the mechanism.

Reply:
Yeah... putting them in the Mechanism seems like it would leak data that should be kept private :)

Reply:
thanks, fixed

Reply:
Is there anything else we can clip out here from the Parameters?  I'm fine with Parameters we need, but wondering if there are others like this that are no longer needed :)

Reply:
>Is there anything else we can clip out here from the Parameters? 

It seems to me that all parameters at now can be used on cross-connect nse side which support wireguard mechanism.

Reply:
I would expect you would still need SrcIP/DstIP and SrcPort/DstPort?

Reply:
I think SrcIP and SrcPort can be removed, but can we at first check this when wireguard will be landed in sdk repos?